samples/infix/reader.cc

```diff
@@ -43,27 +96,62 @@ namespace
     ;
   // clang-format on
 
-  inline const auto wf_operands_tokens = wf_expressions_tokens - Expression;
+  inline const auto wf_tupled_tokens = wf_terminals_tokens | Tuple;
 
   // clang-format off
-  inline const auto wf_pass_trim =
+  inline const auto wf_pass_tuple_literals =
     wf_pass_add_subtract
+    | (Expression <<= wf_tupled_tokens++[1])
+    | (Tuple <<= Expression++)
```

Comment:
Is this right, i.e. what does it mean to be an empty tuple?

Reply:
The usual meaning, as far as I'm aware. Most languages have one. It is never valid to index into it, and appending it to other tuples is the identity function.